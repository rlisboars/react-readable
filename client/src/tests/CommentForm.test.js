import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import ConnectedCommentForm, { CommentForm } from '../components/CommentForm'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-15'
import { store, createStoreWithState, testState } from './mockStore'

const commentData = {
    author: 'thingtwo',
    body: 'Hi there! I am a COMMENT.',
    deleted: false,
    id: '894tuq4ut84ut8v4t8wun89g',
    parentDeleted: false,
    parentId: '8xf0y6ziyjabvozdd253nd',
    timestamp: 1468166872634,
    voteScore: 6
}

const newComment = {
    author: 'newAuthor', 
    body: 'New body!',
    parentId: commentData.parentId
}



describe('CommentForm Component testing', () => {

    const saveCommentFn = jest.fn()
    saveCommentFn.mockReturnValue(Promise.resolve(commentData))
    const setFiltersFn = jest.fn()

    const commentEdit = mount(
        <CommentForm
            comment={commentData}
            parentId={commentData.parentId}
            saveComment={saveCommentFn}
            setFilters={setFiltersFn}
        />
    )

    const commentCreate = mount(
        <CommentForm
            parentId={commentData.parentId}
            saveComment={saveCommentFn}
            setFilters={setFiltersFn}
        />
    )

    const commentCreateThenEdit = mount(
        <CommentForm
            parentId={commentData.parentId}
            saveComment={saveCommentFn}
            setFilters={setFiltersFn}
        />
    )


    it('renders form without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
        <CommentForm
            comment={commentData}
            saveComment={saveCommentFn}
            setFilters={setFiltersFn}
            store={store}
        />, div);
    })

    it('validates edition form of a Comment', () => {
        expect(toJson(commentEdit.find(CommentForm))).toMatchSnapshot()
    })

    it('validates creation form of a Comment', () => {
        expect(toJson(commentCreate.find(CommentForm))).toMatchSnapshot()
    })

    it('validates creation turning into edition', () => {
        commentCreateThenEdit.setProps({ comment: commentData })
        expect(toJson(commentCreateThenEdit.find(CommentForm))).toMatchSnapshot()
    })

    it('save button is working properly for a new register', () => {
        commentCreate.find('textarea').simulate('change', { target: { value: newComment.body }})
        commentCreate.find('input').simulate('change', { target: { value: newComment.author }})
        commentCreate.find('button').at(0).simulate('click')
        expect(saveCommentFn).toHaveBeenCalledWith(
            {
               author: newComment.author,
               body: newComment.body,
               parentId: newComment.parentId
            })
    })

    it('save button is working properly when editing a register', () => {
        commentCreateThenEdit.setProps({ comment: commentData })
        commentCreateThenEdit.find('textarea').simulate('change', { target: { value: newComment.body }})
        commentCreateThenEdit.find('button').at(0).simulate('click')
  
        expect(saveCommentFn).toHaveBeenLastCalledWith(
            {
               body: newComment.body,
               id: commentData.id,
               parentId: commentData.parentId
            })
    })

    it('cancel button is working properly', () => {
       commentEdit.find('button').at(1).simulate('click')
       expect(setFiltersFn).toHaveBeenCalledWith({ isCreatingComment: false })
    })

    it('validates connected CommentForm', () => {
        const testStore = createStoreWithState({
            ...testState, 
            changes: {
                ...testState.changes,
                editingComment: commentData.id
            }
        })
        const connectedCommentForm = mount(
            <Provider store={testStore}>
                    <ConnectedCommentForm 
                        parentId={commentData.parentId}
                    />
            </Provider>
        )
        expect(toJson(connectedCommentForm.find(ConnectedCommentForm))).toMatchSnapshot()
    })

})
