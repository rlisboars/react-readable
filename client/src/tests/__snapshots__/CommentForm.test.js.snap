// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm Component testing validates connected CommentForm 1`] = `
<Connect(CommentForm)
  parentId="8xf0y6ziyjabvozdd253nd"
>
  <CommentForm
    comment={
      Object {
        "author": "thingtwo",
        "body": "Hi there! I am a COMMENT.",
        "deleted": false,
        "id": "894tuq4ut84ut8v4t8wun89g",
        "parentDeleted": false,
        "parentId": "8xf0y6ziyjabvozdd253nd",
        "timestamp": 1468166872634,
        "voteScore": 6,
      }
    }
    filters={
      Object {
        "category": "All",
        "isCreatingComment": false,
        "isCreatingPost": false,
        "sortByComment": "score",
        "sortByPost": "score",
      }
    }
    parentId="8xf0y6ziyjabvozdd253nd"
    saveComment={[Function]}
    setFilters={[Function]}
  >
    <Segment>
      <div
        className="ui segment"
      >
        <Form
          as="form"
        >
          <form
            action={undefined}
            className="ui form"
            onSubmit={[Function]}
          >
            <FormTextArea
              as={[Function]}
              control={[Function]}
              label="Body"
              name="body"
              onChange={[Function]}
              placeholder="Insert your comment"
              required={true}
              value="Hi there! I am a COMMENT."
            >
              <FormField
                control={[Function]}
                label="Body"
                name="body"
                onChange={[Function]}
                placeholder="Insert your comment"
                required={true}
                value="Hi there! I am a COMMENT."
              >
                <div
                  className="required field"
                >
                  <label
                    htmlFor={undefined}
                    key="Body"
                  >
                    Body
                  </label>
                  <TextArea
                    as="textarea"
                    disabled={undefined}
                    name="body"
                    onChange={[Function]}
                    placeholder="Insert your comment"
                    required={true}
                    rows={3}
                    type={undefined}
                    value="Hi there! I am a COMMENT."
                  >
                    <textarea
                      disabled={undefined}
                      name="body"
                      onChange={[Function]}
                      onInput={[Function]}
                      placeholder="Insert your comment"
                      required={true}
                      rows={3}
                      style={
                        Object {
                          "resize": "",
                        }
                      }
                      type={undefined}
                      value="Hi there! I am a COMMENT."
                    />
                  </TextArea>
                </div>
              </FormField>
            </FormTextArea>
            <FormGroup>
              <div
                className="fields"
              >
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  disabled={true}
                  label="Author"
                  name="author"
                  onChange={[Function]}
                  placeholder="Your name"
                  required={false}
                  value="thingtwo"
                  width={3}
                >
                  <FormField
                    control={[Function]}
                    disabled={true}
                    label="Author"
                    name="author"
                    onChange={[Function]}
                    placeholder="Your name"
                    required={false}
                    value="thingtwo"
                    width={3}
                  >
                    <div
                      className="disabled three wide field"
                    >
                      <label
                        htmlFor={undefined}
                        key="Author"
                      >
                        Author
                      </label>
                      <Input
                        disabled={true}
                        name="author"
                        onChange={[Function]}
                        placeholder="Your name"
                        required={false}
                        type="text"
                        value="thingtwo"
                      >
                        <div
                          className="ui disabled input"
                        >
                          <input
                            disabled={true}
                            key="text"
                            name="author"
                            onChange={[Function]}
                            placeholder="Your name"
                            required={false}
                            tabIndex={-1}
                            type="text"
                            value="thingtwo"
                          />
                        </div>
                      </Input>
                    </div>
                  </FormField>
                </FormInput>
                <FormField
                  width={9}
                >
                  <div
                    className="nine wide field"
                  />
                </FormField>
                <FormButton
                  as={[Function]}
                  className="btn-form"
                  control={[Function]}
                  fluid={true}
                  onClick={[Function]}
                  primary={true}
                  width={2}
                >
                  <FormField
                    className="btn-form"
                    control={[Function]}
                    fluid={true}
                    onClick={[Function]}
                    primary={true}
                    width={2}
                  >
                    <div
                      className="two wide field btn-form"
                    >
                      <Button
                        as="button"
                        disabled={undefined}
                        fluid={true}
                        onClick={[Function]}
                        primary={true}
                        required={undefined}
                        type={undefined}
                      >
                        <button
                          className="ui fluid primary button"
                          disabled={undefined}
                          onClick={[Function]}
                          required={undefined}
                          role="button"
                          tabIndex={undefined}
                          type={undefined}
                        >
                          Save
                        </button>
                      </Button>
                    </div>
                  </FormField>
                </FormButton>
                <FormButton
                  as={[Function]}
                  className="btn-form"
                  control={[Function]}
                  fluid={true}
                  onClick={[Function]}
                  width={2}
                >
                  <FormField
                    className="btn-form"
                    control={[Function]}
                    fluid={true}
                    onClick={[Function]}
                    width={2}
                  >
                    <div
                      className="two wide field btn-form"
                    >
                      <Button
                        as="button"
                        disabled={undefined}
                        fluid={true}
                        onClick={[Function]}
                        required={undefined}
                        type={undefined}
                      >
                        <button
                          className="ui fluid button"
                          disabled={undefined}
                          onClick={[Function]}
                          required={undefined}
                          role="button"
                          tabIndex={undefined}
                          type={undefined}
                        >
                          Cancel
                        </button>
                      </Button>
                    </div>
                  </FormField>
                </FormButton>
              </div>
            </FormGroup>
          </form>
        </Form>
      </div>
    </Segment>
  </CommentForm>
</Connect(CommentForm)>
`;

exports[`CommentForm Component testing validates creation form of a Comment 1`] = `
<CommentForm
  parentId="8xf0y6ziyjabvozdd253nd"
  saveComment={[Function]}
  setFilters={[Function]}
>
  <Segment>
    <div
      className="ui segment"
    >
      <Form
        as="form"
      >
        <form
          action={undefined}
          className="ui form"
          onSubmit={[Function]}
        >
          <FormTextArea
            as={[Function]}
            control={[Function]}
            label="Body"
            name="body"
            onChange={[Function]}
            placeholder="Insert your comment"
            required={true}
            value=""
          >
            <FormField
              control={[Function]}
              label="Body"
              name="body"
              onChange={[Function]}
              placeholder="Insert your comment"
              required={true}
              value=""
            >
              <div
                className="required field"
              >
                <label
                  htmlFor={undefined}
                  key="Body"
                >
                  Body
                </label>
                <TextArea
                  as="textarea"
                  disabled={undefined}
                  name="body"
                  onChange={[Function]}
                  placeholder="Insert your comment"
                  required={true}
                  rows={3}
                  type={undefined}
                  value=""
                >
                  <textarea
                    disabled={undefined}
                    name="body"
                    onChange={[Function]}
                    onInput={[Function]}
                    placeholder="Insert your comment"
                    required={true}
                    rows={3}
                    style={
                      Object {
                        "resize": "",
                      }
                    }
                    type={undefined}
                    value=""
                  />
                </TextArea>
              </div>
            </FormField>
          </FormTextArea>
          <FormGroup>
            <div
              className="fields"
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                disabled={false}
                label="Author"
                name="author"
                onChange={[Function]}
                placeholder="Your name"
                required={true}
                value=""
                width={3}
              >
                <FormField
                  control={[Function]}
                  disabled={false}
                  label="Author"
                  name="author"
                  onChange={[Function]}
                  placeholder="Your name"
                  required={true}
                  value=""
                  width={3}
                >
                  <div
                    className="required three wide field"
                  >
                    <label
                      htmlFor={undefined}
                      key="Author"
                    >
                      Author
                    </label>
                    <Input
                      disabled={false}
                      name="author"
                      onChange={[Function]}
                      placeholder="Your name"
                      required={true}
                      type="text"
                      value=""
                    >
                      <div
                        className="ui input"
                      >
                        <input
                          disabled={false}
                          key="text"
                          name="author"
                          onChange={[Function]}
                          placeholder="Your name"
                          required={true}
                          tabIndex={undefined}
                          type="text"
                          value=""
                        />
                      </div>
                    </Input>
                  </div>
                </FormField>
              </FormInput>
              <FormField
                width={9}
              >
                <div
                  className="nine wide field"
                />
              </FormField>
              <FormButton
                as={[Function]}
                className="btn-form"
                control={[Function]}
                fluid={true}
                onClick={[Function]}
                primary={true}
                width={2}
              >
                <FormField
                  className="btn-form"
                  control={[Function]}
                  fluid={true}
                  onClick={[Function]}
                  primary={true}
                  width={2}
                >
                  <div
                    className="two wide field btn-form"
                  >
                    <Button
                      as="button"
                      disabled={undefined}
                      fluid={true}
                      onClick={[Function]}
                      primary={true}
                      required={undefined}
                      type={undefined}
                    >
                      <button
                        className="ui fluid primary button"
                        disabled={undefined}
                        onClick={[Function]}
                        required={undefined}
                        role="button"
                        tabIndex={undefined}
                        type={undefined}
                      >
                        Save
                      </button>
                    </Button>
                  </div>
                </FormField>
              </FormButton>
              <FormButton
                as={[Function]}
                className="btn-form"
                control={[Function]}
                fluid={true}
                onClick={[Function]}
                width={2}
              >
                <FormField
                  className="btn-form"
                  control={[Function]}
                  fluid={true}
                  onClick={[Function]}
                  width={2}
                >
                  <div
                    className="two wide field btn-form"
                  >
                    <Button
                      as="button"
                      disabled={undefined}
                      fluid={true}
                      onClick={[Function]}
                      required={undefined}
                      type={undefined}
                    >
                      <button
                        className="ui fluid button"
                        disabled={undefined}
                        onClick={[Function]}
                        required={undefined}
                        role="button"
                        tabIndex={undefined}
                        type={undefined}
                      >
                        Cancel
                      </button>
                    </Button>
                  </div>
                </FormField>
              </FormButton>
            </div>
          </FormGroup>
        </form>
      </Form>
    </div>
  </Segment>
</CommentForm>
`;

exports[`CommentForm Component testing validates creation turning into edition 1`] = `
<CommentForm
  comment={
    Object {
      "author": "thingtwo",
      "body": "Hi there! I am a COMMENT.",
      "deleted": false,
      "id": "894tuq4ut84ut8v4t8wun89g",
      "parentDeleted": false,
      "parentId": "8xf0y6ziyjabvozdd253nd",
      "timestamp": 1468166872634,
      "voteScore": 6,
    }
  }
  parentId="8xf0y6ziyjabvozdd253nd"
  saveComment={[Function]}
  setFilters={[Function]}
>
  <Segment>
    <div
      className="ui segment"
    >
      <Form
        as="form"
      >
        <form
          action={undefined}
          className="ui form"
          onSubmit={[Function]}
        >
          <FormTextArea
            as={[Function]}
            control={[Function]}
            label="Body"
            name="body"
            onChange={[Function]}
            placeholder="Insert your comment"
            required={true}
            value="Hi there! I am a COMMENT."
          >
            <FormField
              control={[Function]}
              label="Body"
              name="body"
              onChange={[Function]}
              placeholder="Insert your comment"
              required={true}
              value="Hi there! I am a COMMENT."
            >
              <div
                className="required field"
              >
                <label
                  htmlFor={undefined}
                  key="Body"
                >
                  Body
                </label>
                <TextArea
                  as="textarea"
                  disabled={undefined}
                  name="body"
                  onChange={[Function]}
                  placeholder="Insert your comment"
                  required={true}
                  rows={3}
                  type={undefined}
                  value="Hi there! I am a COMMENT."
                >
                  <textarea
                    disabled={undefined}
                    name="body"
                    onChange={[Function]}
                    onInput={[Function]}
                    placeholder="Insert your comment"
                    required={true}
                    rows={3}
                    style={
                      Object {
                        "resize": "",
                      }
                    }
                    type={undefined}
                    value="Hi there! I am a COMMENT."
                  />
                </TextArea>
              </div>
            </FormField>
          </FormTextArea>
          <FormGroup>
            <div
              className="fields"
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                disabled={true}
                label="Author"
                name="author"
                onChange={[Function]}
                placeholder="Your name"
                required={false}
                value="thingtwo"
                width={3}
              >
                <FormField
                  control={[Function]}
                  disabled={true}
                  label="Author"
                  name="author"
                  onChange={[Function]}
                  placeholder="Your name"
                  required={false}
                  value="thingtwo"
                  width={3}
                >
                  <div
                    className="disabled three wide field"
                  >
                    <label
                      htmlFor={undefined}
                      key="Author"
                    >
                      Author
                    </label>
                    <Input
                      disabled={true}
                      name="author"
                      onChange={[Function]}
                      placeholder="Your name"
                      required={false}
                      type="text"
                      value="thingtwo"
                    >
                      <div
                        className="ui disabled input"
                      >
                        <input
                          disabled={true}
                          key="text"
                          name="author"
                          onChange={[Function]}
                          placeholder="Your name"
                          required={false}
                          tabIndex={-1}
                          type="text"
                          value="thingtwo"
                        />
                      </div>
                    </Input>
                  </div>
                </FormField>
              </FormInput>
              <FormField
                width={9}
              >
                <div
                  className="nine wide field"
                />
              </FormField>
              <FormButton
                as={[Function]}
                className="btn-form"
                control={[Function]}
                fluid={true}
                onClick={[Function]}
                primary={true}
                width={2}
              >
                <FormField
                  className="btn-form"
                  control={[Function]}
                  fluid={true}
                  onClick={[Function]}
                  primary={true}
                  width={2}
                >
                  <div
                    className="two wide field btn-form"
                  >
                    <Button
                      as="button"
                      disabled={undefined}
                      fluid={true}
                      onClick={[Function]}
                      primary={true}
                      required={undefined}
                      type={undefined}
                    >
                      <button
                        className="ui fluid primary button"
                        disabled={undefined}
                        onClick={[Function]}
                        required={undefined}
                        role="button"
                        tabIndex={undefined}
                        type={undefined}
                      >
                        Save
                      </button>
                    </Button>
                  </div>
                </FormField>
              </FormButton>
              <FormButton
                as={[Function]}
                className="btn-form"
                control={[Function]}
                fluid={true}
                onClick={[Function]}
                width={2}
              >
                <FormField
                  className="btn-form"
                  control={[Function]}
                  fluid={true}
                  onClick={[Function]}
                  width={2}
                >
                  <div
                    className="two wide field btn-form"
                  >
                    <Button
                      as="button"
                      disabled={undefined}
                      fluid={true}
                      onClick={[Function]}
                      required={undefined}
                      type={undefined}
                    >
                      <button
                        className="ui fluid button"
                        disabled={undefined}
                        onClick={[Function]}
                        required={undefined}
                        role="button"
                        tabIndex={undefined}
                        type={undefined}
                      >
                        Cancel
                      </button>
                    </Button>
                  </div>
                </FormField>
              </FormButton>
            </div>
          </FormGroup>
        </form>
      </Form>
    </div>
  </Segment>
</CommentForm>
`;

exports[`CommentForm Component testing validates edition form of a Comment 1`] = `
<CommentForm
  comment={
    Object {
      "author": "thingtwo",
      "body": "Hi there! I am a COMMENT.",
      "deleted": false,
      "id": "894tuq4ut84ut8v4t8wun89g",
      "parentDeleted": false,
      "parentId": "8xf0y6ziyjabvozdd253nd",
      "timestamp": 1468166872634,
      "voteScore": 6,
    }
  }
  parentId="8xf0y6ziyjabvozdd253nd"
  saveComment={[Function]}
  setFilters={[Function]}
>
  <Segment>
    <div
      className="ui segment"
    >
      <Form
        as="form"
      >
        <form
          action={undefined}
          className="ui form"
          onSubmit={[Function]}
        >
          <FormTextArea
            as={[Function]}
            control={[Function]}
            label="Body"
            name="body"
            onChange={[Function]}
            placeholder="Insert your comment"
            required={true}
            value="Hi there! I am a COMMENT."
          >
            <FormField
              control={[Function]}
              label="Body"
              name="body"
              onChange={[Function]}
              placeholder="Insert your comment"
              required={true}
              value="Hi there! I am a COMMENT."
            >
              <div
                className="required field"
              >
                <label
                  htmlFor={undefined}
                  key="Body"
                >
                  Body
                </label>
                <TextArea
                  as="textarea"
                  disabled={undefined}
                  name="body"
                  onChange={[Function]}
                  placeholder="Insert your comment"
                  required={true}
                  rows={3}
                  type={undefined}
                  value="Hi there! I am a COMMENT."
                >
                  <textarea
                    disabled={undefined}
                    name="body"
                    onChange={[Function]}
                    onInput={[Function]}
                    placeholder="Insert your comment"
                    required={true}
                    rows={3}
                    style={
                      Object {
                        "resize": "",
                      }
                    }
                    type={undefined}
                    value="Hi there! I am a COMMENT."
                  />
                </TextArea>
              </div>
            </FormField>
          </FormTextArea>
          <FormGroup>
            <div
              className="fields"
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                disabled={true}
                label="Author"
                name="author"
                onChange={[Function]}
                placeholder="Your name"
                required={false}
                value="thingtwo"
                width={3}
              >
                <FormField
                  control={[Function]}
                  disabled={true}
                  label="Author"
                  name="author"
                  onChange={[Function]}
                  placeholder="Your name"
                  required={false}
                  value="thingtwo"
                  width={3}
                >
                  <div
                    className="disabled three wide field"
                  >
                    <label
                      htmlFor={undefined}
                      key="Author"
                    >
                      Author
                    </label>
                    <Input
                      disabled={true}
                      name="author"
                      onChange={[Function]}
                      placeholder="Your name"
                      required={false}
                      type="text"
                      value="thingtwo"
                    >
                      <div
                        className="ui disabled input"
                      >
                        <input
                          disabled={true}
                          key="text"
                          name="author"
                          onChange={[Function]}
                          placeholder="Your name"
                          required={false}
                          tabIndex={-1}
                          type="text"
                          value="thingtwo"
                        />
                      </div>
                    </Input>
                  </div>
                </FormField>
              </FormInput>
              <FormField
                width={9}
              >
                <div
                  className="nine wide field"
                />
              </FormField>
              <FormButton
                as={[Function]}
                className="btn-form"
                control={[Function]}
                fluid={true}
                onClick={[Function]}
                primary={true}
                width={2}
              >
                <FormField
                  className="btn-form"
                  control={[Function]}
                  fluid={true}
                  onClick={[Function]}
                  primary={true}
                  width={2}
                >
                  <div
                    className="two wide field btn-form"
                  >
                    <Button
                      as="button"
                      disabled={undefined}
                      fluid={true}
                      onClick={[Function]}
                      primary={true}
                      required={undefined}
                      type={undefined}
                    >
                      <button
                        className="ui fluid primary button"
                        disabled={undefined}
                        onClick={[Function]}
                        required={undefined}
                        role="button"
                        tabIndex={undefined}
                        type={undefined}
                      >
                        Save
                      </button>
                    </Button>
                  </div>
                </FormField>
              </FormButton>
              <FormButton
                as={[Function]}
                className="btn-form"
                control={[Function]}
                fluid={true}
                onClick={[Function]}
                width={2}
              >
                <FormField
                  className="btn-form"
                  control={[Function]}
                  fluid={true}
                  onClick={[Function]}
                  width={2}
                >
                  <div
                    className="two wide field btn-form"
                  >
                    <Button
                      as="button"
                      disabled={undefined}
                      fluid={true}
                      onClick={[Function]}
                      required={undefined}
                      type={undefined}
                    >
                      <button
                        className="ui fluid button"
                        disabled={undefined}
                        onClick={[Function]}
                        required={undefined}
                        role="button"
                        tabIndex={undefined}
                        type={undefined}
                      >
                        Cancel
                      </button>
                    </Button>
                  </div>
                </FormField>
              </FormButton>
            </div>
          </FormGroup>
        </form>
      </Form>
    </div>
  </Segment>
</CommentForm>
`;
